### ISSUE TITLE
Jenkins not restarted after plugin installation if include_role is used

### ISSUE THREAD

**Issue by sebastianludwig** [geerlingguy/ansible-role-jenkins#160]
New in Ansible 2.2 is the [`include_role`](http://docs.ansible.com/ansible/latest/include_role_module.html) module to include a role in another role. If this role is included this way, Jenkins isn't restarted after the plugin installation, because the "restart jenkins" handler is executed once the _including role_ is finished. This is problematic if following tasks rely on the previously installed plugins being activated.

Proposed solution: 
- extract the [Wait for Jenkins to start up](https://github.com/geerlingguy/ansible-role-jenkins/blob/4c63ef29d4a64bbbc46815323ff31485e7bc2c44/tasks/main.yml#L35) task into `wait_for_jenkins.yml` and include it in `main.yml`
- remove the "restart jenkins" handler
- register a variable in [both](https://github.com/geerlingguy/ansible-role-jenkins/blob/4c63ef29d4a64bbbc46815323ff31485e7bc2c44/tasks/plugins.yml#L27) [tasks](https://github.com/geerlingguy/ansible-role-jenkins/blob/4c63ef29d4a64bbbc46815323ff31485e7bc2c44/tasks/plugins.yml#L42) notifying the handler
- Add another task to `plugins.yml` to restart jenkins and also include `wait_for_jenkins.yml`. Both will only be run if one of the registered variables [`.skipped`](http://docs.ansible.com/ansible/latest/common_return_values.html#skipped) is false

@geerlingguy, what's your opinion on this? If you agree, would you accept a PR?

**Comment by sebastianludwig** [comment]
@geerlingguy any thoughts on this?

**Comment by geerlingguy** [comment]
> remove the "restart jenkins" handler

That sounds like a non-starter to me. Basically, this would be adding extra complexity to _this_ role and making it behave in a non-standard way (e.g. handler and notifying a handler) to make other non-standard usages easier (`include_role` is a pretty rough module right now... there are a lot of caveats to its usage).

I'll leave this open in case anyone else might have more thoughts on the matter, but I definitely don't want to try tracking state throughout the role and building tasks with extra conditionals when Ansible's built-in `notify`/handler setup solves that nicely.

**Comment by geerlingguy** [comment]
Could this be related to https://github.com/ansible/ansible/issues/26537 ? Meh, probably not.

**Comment by geerlingguy** [comment]
Or maybe https://github.com/ansible/ansible/issues/26718 ?

**Comment by geerlingguy** [comment]
Basically, I think it would be more fruitful to get Ansible's `include_role` to behave the way you'd like rather than attempt to workaround bugs/missing features by changing all downstream roles.

**Comment by sebastianludwig** [comment]
Yeah, you're probably right 👍 Thanks for taking the time to investigate and reply :-)
